src/felzenszwalb.jl

```diff
@@ -0,0 +1,184 @@
+"""
+```
+index_map, num_segments = felzenszwalb_graph(edges, num_vertices, k)
+```
+
+Efficient Graph-Based Image Segmentation.
+"""
+function felzenszwalb_graph(edges::Array{SVector{3, Float64}}, num_vertices::Int, k::Real)
+
+    num_edges = length(edges)
+    num_sets = num_vertices
+    rank = zeros(num_sets)
+    set_size = ones(num_sets)
+    parent = Array(range(1, num_sets))
+    threshold = fill(k, num_vertices)
+    
+    function find_root(x)
+        x = Int(x)
+        if parent[x]!=x
+            parent[x] = find_root(parent[x])
+        end
+        return parent[x]
+    end
+
+    function merge_trees(x, y)
+        x = Int(x)
+        y = Int(y)
+        num_sets -= 1
+        if rank[x]<rank[y]
+            parent[x] = y
+            set_size[y] += set_size[x]
+            return y
+        elseif rank[x]>rank[y]
+            parent[y] = x
+            set_size[x] += set_size[y]
+            return x
+        else
+            parent[y] = x
+            set_size[x] += set_size[y]
+            rank[x] += 1
+            return x
+        end
+    end
+
+    sort(edges, lt = (x,y)->(x[3]<y[3]))
```

Comment:
Should this be `sort!`? Otherwise I don't think this does anything, since you're not capturing the output.